[Feat] #13: 마이페이지 CRUD 기능구현

WooSeok77 opened the pull request:
## 🧑‍💻 요약
- 마이페이지 기능구현

## ✅ 작업 내용
- [x] 마이페이지 메인
- [ ] 상대방이 보는 마이페이지
- [x] 마이페이지 프로필 수정
- [x] 친구목록(팔로워, 팔로우)

## 참고 사항
-메인은 현재 하단 리스트를 제외하고 상단부기능만 구현했습니다(nickname, bio, job, position, followercount, followingcount)
-피그마상 상대방이 보는마이페이지도 별반다르지 않지만(들어가는 버튼만달라짐) 동일한 api를 사용해도 될지 고민입니다.(해서 해당 작업체크는 애매해서 비워놨습니다.)
-팔로워,팔로우 리스트가 나오게끔 구현했습니다만 한번 확인이 필요할꺼같습니다.
## 관련이슈
#13

WooSeok77 committed:
[#13] feat: 프로필 수정기능 구현

WooSeok77 committed:
[#13] feat: 마이페이지메인 상단기능 구현

WooSeok77 committed:
[#13] feat: 마이페이지메인 상단 팔로워, 팔로우 수 구현

WooSeok77 committed:
[#13] feat: 팔로우, 팔로워 기능구현

WooSeok77 committed:
[#13] chore: 주석제거

sohn919 commented on src/main/java/fc/server/palette/member/controller/MemberController.java:
```diff
@@ -0,0 +1,78 @@
+
+
+    @GetMapping("/mypage/{memberId}")
+    public ResponseEntity<?> mypage (@PathVariable Long memberId){
```
메서드의 파라미터가 어떤 부분은 `메서드 명(param)` 이고 어떤 부분에서는 `메서드 명 (param)` 형태입니다. 
현재 프로젝트에서는 다들 붙여 쓰시는데 한 번 확인해보시는 것도 좋을 것 같습니다!

WooSeok77 replied:
넵 확인하겠습니다!

sohn919 commented on src/main/java/fc/server/palette/member/controller/MemberController.java:
```diff
@@ -0,0 +1,78 @@
+        memberService.unfollow(followedId, followingId);
+        return ResponseEntity.ok("언팔로우완료");
+    }
+
+
+
+
+
+
+
```
전체적으로 의미 없는 공백 부분들이 많이 있습니다.
이런 부분을 신경 써주시면 좋을 것 같아요!

WooSeok77 replied:
공백 고쳐보겠습니다

sohn919 commented on src/main/java/fc/server/palette/member/service/MemberService.java:
```diff
@@ -0,0 +1,157 @@
+
+    @Transactional
+    public void updateMember(Long loginMemberId, Long memberId, MemberProfileDto memberProfileDto){
+        Member member = memberRepository.findById(memberId).orElseThrow(
+                () -> new Exception400(memberId.toString(), ExceptionMessage.NO_MEMBER_ID));
```
저희가 orElseThrow 한 칸 내리기로 했었는데 한 번 확인해주세요

sohn919 commented on src/main/java/fc/server/palette/member/service/MemberService.java:
```diff
@@ -0,0 +1,157 @@
+
+    public MemberMyPageDto myPageInfo(Long memberId) {
+        Optional<Member> memberOptional = Optional.ofNullable(memberRepository.findById(memberId).orElseThrow(
+                () -> new Exception400(memberId.toString(), ExceptionMessage.NO_MEMBER_ID)));
```
`null` 일 경우 `orElseThrow` 로 던졌는데 앞에서 `Optional.ofNullable` 을 해서 `null` 을 허용하게 한다는 부분에서 이해가 안되는 부분인데 설명 해주실 수 있을까요?

sohn919 commented on src/main/java/fc/server/palette/member/dto/response/FollowInfoDto.java:
```diff
@@ -0,0 +1,16 @@
+
+@Getter
+@Builder
+@NoArgsConstructor
```
여기에 기본 생성자가 필요한지 한 번 확인해주세요..!

sohn919 commented on src/main/java/fc/server/palette/member/dto/request/MemberProfileDto.java:
```diff
@@ -0,0 +1,30 @@
+
+@Getter
+@NoArgsConstructor
+@AllArgsConstructor
```
여기에 `@AllArgsConstructor` 이 필요한 것인지 한 번 확인해주세요..!

WooSeok77 committed:
[#13] fix: MemberService내 myPageInfo 메서드 수정 및 기타수정

피드백에서 FollowInfoDto의 경우 MemberService내 정의된 followInfo메서드의 빌더패턴에 의해 기본생성자및 전체생성자가 필요합니다.

brotherstone97 commented on src/main/java/fc/server/palette/member/controller/MemberController.java:
모든 요청url에 member가 들어간다면
18번째 라인에 @RequestMapping의 value로 /api/member라고 하셔도 될 것 같습니다.

brotherstone97 commented on src/main/resources/application-dev.yml:
application-dev.yml파일 아직도 제외처리 안되고 있는 것 같습니다..!

brotherstone97 commented on src/main/java/fc/server/palette/member/entity/Member.java:
혹시 fetchType을 LAZY에서 EAGER로 바꾸신 이유가 있으실까요??
성능은 lazy가 더 우수한 걸로 알고있는데

WooSeok77 replied:
프로젝트 실행해서 postman상으로 테스트해보려고했을시 관련오류가 생겨서 EAGER로 변경했었습니다..!

brotherstone97 replied:
LAZY방식이 현재 필요한 데이터만 db에서 불러오기 때문이라고 생각합니다., 
그에 대한 해결책으로 필요한 데이터를 dto로 감싸 db에서 불러오면 될것으로 생각됩니다.
관련 내용 정리한 노션링크 남겨드릴게요
https://www.notion.so/FetchType-LAZY-839baa741dd84687b4962518d7576a03?pvs=4
